src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -718,6 +718,17 @@ private bool AreNamedTypesEqual(NamedTypeSymbol type, NamedTypeSymbol other)
                 // TODO: Test with overloads (from PE base class?) that have modifiers.
                 Debug.Assert(!type.HasTypeArgumentsCustomModifiers);
                 Debug.Assert(!other.HasTypeArgumentsCustomModifiers);
+
+                if (type.IsTupleType)
+                {
+                    if (!other.IsTupleType)
+                    {
+                        return false;
+                    }
+
+                    type.TupleElementTypes.SequenceEqual(other.TupleElementTypes, AreTypesEqual);
```

Comment:
Thanks for catching this. 
This bothers me, as it means the test that covers this code is inadequate (it's only covering the change above, at line 492). I'll try to cover this better.